components/Map/index.js

```diff
@@ -0,0 +1,117 @@
+import Link from "next/link";
+
+import styled from "styled-components";
+
+import { MapContainer, Marker, Popup, TileLayer } from "react-leaflet";
+import Lottie from "lottie-react";
+import * as L from "leaflet";
+import "leaflet/dist/leaflet.css";
+import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css";
+import "leaflet-defaulticon-compatibility";
+
+import LoadingAnimation from "@/public/lottiefiles/LoadingAnimation.json";
+
+import CurrentMarker from "@/components/Map/CurrentMarker";
+
+const StyledMapAndIconContainer = styled.div`
+  position: relative;
+`;
+const StyledMapContainer = styled(MapContainer)`
+  height: 30vh;
+  width: calc(100%+30px);
+  margin: 0 -15px;
+  z-index: 0;
+`;
+
+const StyledLink = styled(Link)`
+  h2 {
+    color: var(--primaryDarkColor);
+    &:hover {
+      cursor: pointer;
+      color: var(--accentColor);
+    }
+  }
+  text-decoration: none;
+`;
+
+const StyledLottie = styled(Lottie)`
+  width: 30%;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  z-index: 2;
+`;
+const greyIcon = new L.Icon({
+  iconUrl:
+    "https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-gold.png",
+  shadowUrl:
+    "https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png",
+  iconSize: [17, 27],
+  iconAnchor: [8, 27],
+  popupAnchor: [1, -22],
+  shadowSize: [27, 27],
+});
+
+export default function Map({
+  stores,
+  currentStore = {},
+  currentCoordinates,
+  isLoading,
+}) {
+  const isCreateStore = Object.keys(currentStore).length === 0;
+  return (
+    <StyledMapAndIconContainer>
+      {isLoading && (
+        <StyledLottie loop={true} animationData={LoadingAnimation} />
+      )}
+      <StyledMapContainer
+        center={[51.51365366910267, 7.469919177246061]}
```

Comment:
We already talked about this, I would not center the map over Dortmund initially, its better to center it over the user-location.

Reply:
You're totally right. We wrote a separate component, that get's the current user location and set the center of the map and the `setView()` hook to the current user location. When creating a new store, that hasn't a location, yet, since the user is still in the `create` Store - Form, the map also just shows the current user location. 
Thanks for your help and recommendations 🙏